File: sourcecode/client/src/hci/mainDialog/assets/toolbarPane.cpp
//  Copyright 2014-2015 Integrated Test Management Centre Foundation Team
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
// http://www.apache.org/licenses/LICENSE-2.0
//////////////////////////////////////////////////////////////////////////
#include <hci/mainDialog/assets/toolbarPane.h>


wxWindowID ID_treeCtrl = ID_toolbarPane + 0x01;


ToolbarPane::ToolbarPane( wxWindow *parent
                        , const wxSize &size )
            : wxPanel( parent
                     , ID_toolbarPane
                     , wxDefaultPosition
                     , size )
{
    wxBoxSizer *topLevelSizer = new wxBoxSizer( wxHORIZONTAL );

    m_treeCtrl = new AssetsTreeCtrl( this
                                   , ID_treeCtrl
                                   , size );

    topLevelSizer->Add( m_treeCtrl
                      , wxEXPAND
                      , wxEXPAND );

    SetSizerAndFit( topLevelSizer );  
}


ToolbarPane::~ToolbarPane()
{
    if ( m_treeCtrl ) delete m_treeCtrl;
}


void ToolbarPane::AddAssetItem( AssetDataTransaction *transaction )
{
    m_treeCtrl->AddAssetItem( transaction->GetID()
                            , transaction->GetParentID()
                            , transaction->GetTitle()
                            , transaction->GetAssetData()->GetType() );
}

File: src/Client/HCI/MainDialog/AssetsPanel/ToolbarPane.py
#  Copyright 2014-2016 Integrated Test Management Centre Foundation Team
#  Licensed under the Apache License, Version 2.0 (the "License");
#  you may not use this file except in compliance with the License.
#  You may obtain a copy of the License at
#  http://www.apache.org/licenses/LICENSE-2.0
#  //////////////////////////////////////////////////////////////////////////
import wx
from AssetsTreeCtrl import *
#include <dataModels/assetDataTransaction.h>


#const wxWindowID ID_toolbarPane = 0x010;

# wxID_ANY

class ToolbarPane(wx.Panel):

	def __init__(self, parent, size):
		super(ToolbarPane, self).__init__(parent, wx.ID_ANY, 
				wx.DefaultPosition, size)

		self._topLevelSizer = wx.BoxSizer(wx.HORIZONTAL)

		self._treeCtrl = AssetsTreeCtrl(self, size)

		self._topLevelSizer.Add(self._treeCtrl, wx.EXPAND, wx.EXPAND)

		self.SetSizerAndFit(self._topLevelSizer)

"""
	void ToolbarPane::AddAssetItem( AssetDataTransaction *transaction )

		m_treeCtrl->AddAssetItem( transaction->GetID()
                            , transaction->GetParentID()
                            , transaction->GetTitle()
                            , transaction->GetAssetData()->GetType() );
"""
